src/Frontend/Path.hs

```diff
@@ -174,35 +176,39 @@ listTopicIdeas :: Topic -> Main
 
 ideaMode :: IdeaMode -> UriPath -> UriPath
-ideaMode ListIdeas                      root = root </> "ideas"
-ideaMode (ViewIdea i)                   root = root </> "idea" </> uriPart i </> "view"
-ideaMode (EditIdea i)                   root = root </> "idea" </> uriPart i </> "edit"
-ideaMode (LikeIdea i)                   root = root </> "idea" </> uriPart i </> "like"
-ideaMode (VoteIdea i v)                 root = root </> "idea" </> uriPart i </> "vote"
-                                                    </> uriPart v
-ideaMode (JudgeIdea i v)                root = root </> "idea" </> uriPart i </> "jury"
-                                                    </> uriPart v
-ideaMode (CommentIdea i)                root = root </> "idea" </> uriPart i </> "comment"
-ideaMode (OnComment ctx c m)            root = commentMode ctx c m root
-ideaMode CreateIdea                     root = root </> "idea" </> "create"
+ideaMode ListIdeas         root = root </> "ideas"
+ideaMode (ViewIdea i)      root = root </> "idea" </> uriPart i </> "view"
+ideaMode (EditIdea i)      root = root </> "idea" </> uriPart i </> "edit"
+ideaMode (LikeIdea i)      root = root </> "idea" </> uriPart i </> "like"
+ideaMode (VoteIdea i v)    root = root </> "idea" </> uriPart i </> "vote"
+                                       </> uriPart v
+ideaMode (JudgeIdea i v)   root = root </> "idea" </> uriPart i </> "jury"
+                                       </> uriPart v
+ideaMode (CommentIdea i)   root = root </> "idea" </> uriPart i </> "comment"
+ideaMode (OnComment cid m) root = commentMode cid m root
+ideaMode CreateIdea        root = root </> "idea" </> "create"
 
 commentAnchor :: IsString s => AUID Comment -> s
 commentAnchor (AUID c) = fromString $ "comment-" <> show c
 
-commentMode :: CommentContext -> AUID Comment -> CommentMode -> UriPath -> UriPath
-commentMode (CommentContext idea mc) c m root =
+commentMode :: CommentId -> CommentMode -> UriPath -> UriPath
+commentMode (CommentId _loc i parents commentId) m root =
     case m of
-        ReplyComment  -> base </> "reply"
-        DeleteComment -> base </> "delete"
-        ReportComment -> base </> "report"
-        ViewComment   -> root </> "idea" </> uriPart i </> "view" </#> commentAnchor c
-        VoteComment v -> base </> "vote" </> uriPart v
+        ReplyComment  -> base 1 </> "reply"
+        DeleteComment -> base 2 </> "delete"
+        ReportComment -> base 2 </> "report"
+        VoteComment v -> base 2 </> "vote" </> uriPart v
+        ViewComment   -> root  </> "idea" </> uriPart i </> "view" </#> commentAnchor commentId
   where
-    i = idea ^. _Id
-    base =
-        case mc of
-            Nothing -> root </> "idea" </> uriPart i </> "comment" </> uriPart c
-            Just p  -> root </> "idea" </> uriPart i </> "comment" </> uriPart (p ^. _Id) </> "reply" </> uriPart c
+    -- FIXME: Deep replies are not supported yet.
```

Comment:
If "deep replies" means "comment hierarchies of depth >2": Leave the sentence, but drop the FIXME?  We are not supposed to support that.